src/DETHRACE/common/main.c

```diff
@@ -27,7 +29,21 @@ void QuitGame() {
         DoDemoGoodbye();
     }
 
-    NOT_IMPLEMENTED();
+    gProgram_state.racing = 0;
+    SaveOptions();
+    if (gNet_mode != eNet_mode_none) {
+        NetLeaveGame(gCurrent_net_game);
+    }
+    ShutdownNetIfRequired();
+    if (gSound_available) {
+        DRS3ShutDown();
+    }
+    if (gBr_initialized) {
+        ClearEntireScreen();
+    }
+    PDRevertPalette();
+    StopMusic();
+    PDShutdownSystem();
```

Comment:
I can also see `CloseDiagnostics()` and `exit(0)`, even though `PDShutdownSystem()` on Windows will already exit the system.
But I'm not sure
```suggestion
    PDShutdownSystem();
    CloseDiagnostics();
    exit(0);
```

Reply:
good spotting! For some reason, IDA was confused by this function and the disassembler was missing the last bit of it. Good reminder to not be lazy and double check the assembly too :)

Reply:
In that regard, ghidra is moody too.
It does not always show what happens after `FatalError`, but the decompiler showed everything here.
I suppose you need to always be careful when marking a function as "does not return".